pkg/storage/chunk/cache/memcached.go

```diff
@@ -11,14 +11,20 @@ import (
 	"github.com/go-kit/log"
 	instr "github.com/grafana/dskit/instrument"
 	"github.com/grafana/gomemcache/memcache"
+	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
+	"go.uber.org/atomic"
 
 	"github.com/grafana/loki/pkg/logqlmodel/stats"
 	"github.com/grafana/loki/pkg/util/constants"
 	"github.com/grafana/loki/pkg/util/math"
 )
 
+var (
+	ErrMemcachedClosed = errors.New("memcached is closed")
```

Comment:
`memcached is closed` is not a very helpful error message. Is the client closed? Is the server down?
Please clarify this.

Reply:
Done